### PR TITLE
feat: add support for `none` to indicate no regularization

### PR DESCRIPTION
**Pull request by nakul-krishnakumar**
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown_pkg_readmes status: passed
  - task: lint_markdown_docs status: na
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #None.

## Description

> What is the purpose of this pull request?

This pull request:

-    Adds `none` value for `penalties` enum.

## Related Issues

> Does this pull request have any related issues?

This pull request has the following related issues:

-   None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

> When authoring the changes proposed in this PR, did you use any kind of AI assistance?

-   [ ] Yes
-   [x] No

If you answered "yes" above, how did you use AI assistance?

-   [ ] Code generation (e.g., when writing an implementation or fixing a bug)
-   [ ] Test/benchmark generation
-   [ ] Documentation (including examples)
-   [ ] Research and understanding

#### Disclosure

> If you answered "yes" to using AI assistance, please provide a short disclosure indicating how you used AI assistance. This helps reviewers determine how much scrutiny to apply when reviewing your contribution. Example disclosures: "This PR was written primarily by Claude Code." or "I consulted ChatGPT to understand the codebase, but the proposed changes were fully authored manually by myself.".

{{TODO: add disclosure if applicable}}

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by nakul-krishnakumar** [comment]
This PR currently blocks #13669 and #13670

File: lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/docs/repl.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@
     - l2: L2 regularization (also called ridge regression) encourages smaller,
     more evenly distributed weights by adding a penalty based on the square of
     the coefficients.
+    - none: No regularization.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/docs/repl.txt]:
```suggestion
    - none: no regularization.
```

File: lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/lib/enum.js [hunk 2]
```diff
@@ -44,7 +44,10 @@ function enumerated() {
 		'l1': 1,
 
 		// L2 regularization (also called ridge regression) encourages smaller, more evenly distributed weights by adding a penalty based on the square of the coefficients:
-		'l2': 2
+		'l2': 2,
+
+		// No regularization
```
Comment (kgryte) [inline lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/lib/enum.js]:
```suggestion
		// No regularization:
```

File: lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/README.md [hunk 3]
```diff
@@ -46,14 +46,15 @@ Returns a list of SGD classification penalties.
...
-// e.g., returns [ 'elasticnet', 'l1', 'l2' ]
+// e.g., returns [ 'elasticnet', 'l1', 'l2', 'none' ]
 ```
 
 The output array contains the following penalties:
 
 -   `elasticnet`: regularization method that linearly combines the L1 and L2 penalties of the lasso and ridge methods.
 -   `l1`: L1 regularization (also called LASSO) leads to sparse models by adding a penalty based on the absolute value of coefficients.
 -   `l2`: L2 regularization (also called ridge regression) encourages smaller, more evenly distributed weights by adding a penalty based on the square of the coefficients.
+-   `none`: No regularization.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/README.md]:
```suggestion
-   `none`: no regularization.
```

File: lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/README.md [hunk 4]
```diff
@@ -128,6 +129,7 @@ An enumeration of SGD classification penalties with the following fields:
 -   **STDLIB_ML_SGD_CLASSIFICATION_ELASTICNET**: regularization method that linearly combines the L1 and L2 penalties of the lasso and ridge methods.
 -   **STDLIB_ML_SGD_CLASSIFICATION_L1**: L1 regularization (also called LASSO) leads to sparse models by adding a penalty based on the absolute value of coefficients.
 -   **STDLIB_ML_SGD_CLASSIFICATION_L2**: L2 regularization (also called ridge regression) encourages smaller, more evenly distributed weights by adding a penalty based on the square of the coefficients.
+-   **STDLIB_ML_SGD_CLASSIFICATION_NONE**: No regularization.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/README.md]:
```suggestion
-   **STDLIB_ML_SGD_CLASSIFICATION_NONE**: no regularization.
```

File: lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/README.md [hunk 5]
```diff
@@ -128,6 +129,7 @@ An enumeration of SGD classification penalties with the following fields:
 -   **STDLIB_ML_SGD_CLASSIFICATION_ELASTICNET**: regularization method that linearly combines the L1 and L2 penalties of the lasso and ridge methods.
 -   **STDLIB_ML_SGD_CLASSIFICATION_L1**: L1 regularization (also called LASSO) leads to sparse models by adding a penalty based on the absolute value of coefficients.
 -   **STDLIB_ML_SGD_CLASSIFICATION_L2**: L2 regularization (also called ridge regression) encourages smaller, more evenly distributed weights by adding a penalty based on the square of the coefficients.
+-   **STDLIB_ML_SGD_CLASSIFICATION_NONE**: no regularization.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/ml/base/sgd-classification/penalties/README.md:132]: This naming convention is a bit risky insofar as this doesn't clash with any other enum belonging to other parameterizations (e.g., loss functions, learning rates, etc).

Fine to leave as is for now. Otherwise, we may need to update our conventions if this assumption is invalidated in the future.

**Review by kgryte — APPROVED** [review]
LGTM after clean-up.
